java/vector/src/test/java/org/apache/arrow/vector/TestDictionaryEncodedVector.java

```diff
@@ -0,0 +1,132 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector;
+
+import static java.nio.channels.Channels.newChannel;
+import static java.util.Arrays.asList;
+import static org.apache.arrow.vector.TestUtils.newVarCharVector;
+
+import java.io.*;
```

Comment:
This should be a style problem.

Reply:
Agreed, I wonder if this PR is old enough that we didn't have checkstyle enforcement?

Reply:
@emkornfield , I have prepared a [doc](https://docs.google.com/document/d/1Y2E6RbZkUj3SwuEJrlEjaeIPmCA1SIsi9wmbJmVlB2I/edit?usp=sharing) for the dictionary related use cases. 

This PR solves problem 2 in the doc (misleading constructor), and it is a relatively small problem and involves small changes. 

Can we first merge this PR  and close this issue? It looks to me except some style problems, and some comments need to be removed.